File: cpp/foxford_2nd_year/17/4_foxford.cpp
#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;
int main()
{
    int n, m, i;
    long long ans = 0;
    cin >> n >> m;
    vector<int> a(n);
    for (i = 0; i < n; ++i)
        cin >> a[i];
    sort(a.begin(), a.end());
    for (i = 0; i < n; ++i)
    {
        int j = upper_bound(a.begin(), a.end(), a[i] + m) - a.begin();
        ans += j - 1 - i;
    }
    cout << ans << endl;
    return 0;
}

File: cpp/foxford_2nd_year/17/4_foxford.py
n, r = map(int, input().split())
A = list(map(int, input().split()))
A.sort()
i = 0
j = 0
ans = 0
for i in range(len(A)):
    while j < len(A) and A[j] - A[i] <= r:
        j+=1
    if (j < n) or (j == n and A[n - 1] - A[i] <= r):
        ans += j - 1 - i   
print(ans)